src/robot/model/body.py

```diff
@@ -120,39 +136,39 @@ def create(self):
             f"Use item specific methods like 'create_keyword' instead."
         )
 
-    def create_keyword(self, *args, **kwargs):
+    def create_keyword(self, *args, **kwargs) -> 'Keyword':
         return self._create(self.keyword_class, 'create_keyword', args, kwargs)
 
-    def _create(self, cls, name, args, kwargs):
+    def _create(self, cls: 'Type[C]|None', name: str, args, kwargs) -> C:
         if cls is None:
             raise TypeError(f"'{full_name(self)}' object does not support '{name}'.")
```

Comment:
I don't think `cls` here should have `None` as a valid type. Passing `None` here would be a bug and we even raise a TypeError in that case.

`args` and `kwargs` should have typing like `Iterable` and `Mapping`, respectively.

Reply:
I added the None, as the type checker complains about the `xxx_class` Class attributes Since they are initialised with `None` . 

![image](https://user-images.githubusercontent.com/28574129/236680585-ac147af8-7a9d-46bc-8e69-db79e70c94eb.png)
 
Not accepting `None` here causes all of the `create_xxx` keywords to report typing errors.

Reply:
Alternatively, I can do the following, and remove `None`

![image](https://user-images.githubusercontent.com/28574129/236680782-71e43c4c-85a8-481e-9488-360f25cbcde0.png)

Reply:
Somewhat annoying to have such big `if/else`, but it's probably better than the other alternative. I can merge it but we can try to come up with something better later.